### PR TITLE
test(perf): add regression tests for LoadTestUsingSMPHandler boundary checks

### PR DESCRIPTION
**Pull request by Maanvi212006**
**Notes for Reviewers**
Adds regression tests for the two bounds-checks in `LoadTestUsingSMPHandler` 
that guard against runtime panics on malformed SMP payloads:

- Empty `clients` array → 400 Bad Request
- Empty `endpointUrls` array → 400 Bad Request

Related to #18979 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: server/handlers/load_test_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	req := httptest.NewRequest(http.MethodPost, "/perf/profile", body)
+	req.Header.Set("Content-Type", "application/json")
+	w := httptest.NewRecorder()
+
+	handler.LoadTestUsingSMPHandler(w, req, nil, nil, nil)
+
+	if w.Code != http.StatusBadRequest {
+		t.Errorf("expected 400 for empty endpointUrls, got %d", w.Code)
+	}
+}
```
[the remark this answers is not in the document]
Comment (Maanvi212006) [reply server/handlers/load_test_handler_test.go]: accepted

File: server/handlers/load_test_handler_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+	req := httptest.NewRequest(http.MethodPost, "/perf/profile", body)
+	req.Header.Set("Content-Type", "application/json")
+	w := httptest.NewRecorder()
+
+	handler.LoadTestUsingSMPHandler(w, req, nil, nil, nil)
+
+	if w.Code != http.StatusBadRequest {
+		t.Errorf("expected 400 for empty clients, got %d", w.Code)
+	}
```
[the remark this answers is not in the document]
Comment (Maanvi212006) [reply server/handlers/load_test_handler_test.go]: The suggested change keeps both t.Errorf and t.Fatalf — the Errorf line becomes unreachable dead code since Fatalf stops execution immediately.

File: server/handlers/load_test_handler_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+	req := httptest.NewRequest(http.MethodPost, "/perf/profile", body)
+	req.Header.Set("Content-Type", "application/json")
+	w := httptest.NewRecorder()
+
+	handler.LoadTestUsingSMPHandler(w, req, nil, nil, nil)
+
+	if w.Code != http.StatusBadRequest {
+		t.Errorf("expected 400 for empty endpointUrls, got %d", w.Code)
+	}
```
[the remark this answers is not in the document]
Comment (Maanvi212006) [reply server/handlers/load_test_handler_test.go]: The suggested change keeps both t.Errorf and t.Fatalf — the Errorf line becomes unreachable dead code since Fatalf stops execution immediately.

**Comment by Maanvi212006** [comment]
@jamieplu all tests are passing. Ready for review
